### PR TITLE
@sweir27 => Purchase 106 QA Fixes

### PR DESCRIPTION
**Pull request by erikdstock**
This PR addresses design feedback from @briansw in PURCHASE-128, -129 AND -130:
https://artsyproduct.atlassian.net/browse/PURCHASE-106

In order to meet PURCHASE-129 (hover on link should not trigger checkbox hover state) I had to unset some of the default `.artsy-checkbox` styling and wrap the input in its text label, then align the link with the text. It appears fine for me in mobile and desktop, but it did take some nudging.

registerform

EDIT: Update with new styling that fixes the period issue pointed out by @briansw below. It adds a little extra javascript and an `a::before` pseudoelement to block the hover state when hovering on the link specifically.
registerform

### CODE REVIEW

File: src/mobile/apps/auction_support/stylesheets/index.styl [hunk 1]
```diff
@@ -153,3 +136,40 @@ checkbox-size = 20px
   display block
   input
     padding 10px 0 11px 0
+
+.registration-form-section.accept-cos
```
Comment (sweir27) [inline src/mobile/apps/auction_support/stylesheets/index.styl]: This feels like another thing we could consider abstracting out, especially if it's going to be copied again. But we can evaluate during the upcoming stories.

Comment (erikdstock) [reply src/mobile/apps/auction_support/stylesheets/index.styl]: I agree- there are some differences though (`red-color` vs `bad-red-color` comes to mind)

**Comment by briansw** [comment]
@erikdstock @sweir27 the period isn't sitting correctly on the baseline quite yet.

File: src/desktop/apps/auction_support/stylesheets/index.styl [hunk 2]
```diff
@@ -42,19 +42,35 @@
       display flex
       justify-content center
       align-content center
+      // Vertically align the period.
+      line-height 1.5em
```
Comment (sweir27) [inline src/desktop/apps/auction_support/stylesheets/index.styl]: Why do we need these separate `line-height`s for the text and the link?

Comment (sweir27) [inline src/desktop/apps/auction_support/stylesheets/index.styl]: Why do we need these separate `line-height`s for the text and the link?

Comment (erikdstock) [reply src/desktop/apps/auction_support/stylesheets/index.styl]: Gone now. I was struggling with line heights to make the link and period appear on the same line.

Comment (erikdstock) [reply src/desktop/apps/auction_support/stylesheets/index.styl]: They are gone now! Good riddance!

File: src/mobile/apps/auction_support/stylesheets/index.styl (hunk 1 above)
Comment (sweir27) [reply src/mobile/apps/auction_support/stylesheets/index.styl]: That could probably be overridden in a specific case if all other styles are being inherited. But we can consolidate in a future PR!

File: src/desktop/apps/auction_support/client/registration_form.coffee [hunk 3]
```diff
@@ -121,10 +121,12 @@ module.exports = class RegistrationForm extends ErrorHandlingForm
   validateAcceptCOS: (e) ->
     if @$acceptCOS.prop('checked')
       @$('.artsy-checkbox').removeClass('error')
+      @$('#cos_link').removeClass('conceal-hover-state')
```
Comment (sweir27) [inline src/desktop/apps/auction_support/client/registration_form.coffee]: Is there any way to achieve the hover behavior without needing jquery?

Comment (erikdstock) [reply src/desktop/apps/auction_support/client/registration_form.coffee]: Updated to use z-index.

**Pull request merged**